pkg/config/config_test.go

```diff
@@ -389,3 +390,93 @@ func createOrdererStandardPolicies() map[string]*Policy {
 
 	return policies
 }
+
+func TestComputeUpdate(t *testing.T) {
+	gt := NewGomegaWithT(t)
+
+	value1Name := "foo"
+	value2Name := "bar"
+	base := common.Config{
+		ChannelGroup: &common.ConfigGroup{
+			Version: 7,
+			Values: map[string]*common.ConfigValue{
+				value1Name: {
+					Version: 3,
+					Value:   []byte("value1value"),
+				},
+				value2Name: {
+					Version: 6,
+					Value:   []byte("value2value"),
+				},
+			},
+		},
+	}
+	updated := common.Config{
+		ChannelGroup: &common.ConfigGroup{
+			Values: map[string]*common.ConfigValue{
+				value1Name: base.ChannelGroup.Values[value1Name],
+				value2Name: {
+					Value: []byte("updatedValued2Value"),
+				},
+			},
+		},
+	}
+
+	channelID := "testChannel"
+
+	expectedReadSet := newConfigGroup()
+	expectedReadSet.Version = 7
+
+	expectedWriteSet := newConfigGroup()
+	expectedWriteSet.Version = 7
+	expectedWriteSet.Values = map[string]*common.ConfigValue{
+		value2Name: {
+			Version: 7,
+			Value:   []byte("updatedValued2Value"),
+		},
+	}
+
+	expectedConfig := common.ConfigUpdate{
+		ChannelId: channelID,
+		ReadSet:   expectedReadSet,
+		WriteSet:  expectedWriteSet,
+	}
+
+	configUpdate, err := ComputeUpdate(&base, &updated, channelID)
+	gt.Expect(err).NotTo(HaveOccurred())
+	gt.Expect(configUpdate).To(Equal(&expectedConfig))
+
+}
+
+func TestComputeUpdateFailures(t *testing.T) {
+	t.Parallel()
+
+	base := common.Config{}
+	updated := common.Config{}
+
+	for _, test := range []struct {
+		name        string
+		channelID   string
+		expectedErr string
+	}{
+		{
```

Comment:
Let's make sure we always include the field name instead of relying on the order of the fields.

Reply:
🛠